pennylane/beta/tapes/cv_param_shift.py

```diff
@@ -0,0 +1,458 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+CV parameter shift quantum tape.
+
+Provides analytic differentiation for variational circuits with parametrized Gaussian CV gates
+and first- and second-order observables.
+"""
+# pylint: disable=attribute-defined-outside-init,too-many-branches
+import itertools
+import warnings
+
+import numpy as np
+
+import pennylane as qml
+from pennylane.beta.queuing import MeasurementProcess
+
+from .qubit_param_shift import QubitParamShiftTape
+
+
+class CVParamShiftTape(QubitParamShiftTape):
+    r"""Quantum tape for CV parameter-shift analytic differentiation method.
+
+    This class extends the :class:`~.jacobian` method of the quantum tape
+    to support analytic gradients of Gaussian CV operations using the parameter-shift rule.
+    This gradient method returns *exact* gradients, and can be computed directly
+    on quantum hardware. Simply pass ``method=analytic`` when computing the Jacobian:
+
+    >>> tape.jacobian(dev, method="analytic")
+
+    For more details on the quantum tape, please see :class:`~.QuantumTape`.
+
+    This tape supports analytic gradients of photonic circuits that satisfy
+    the following constraints with regards to measurements:

```

Comment:
We noticed during HOGS a couple of weeks ago that we never define in PL _when_ the CV parameter-shift rule is valid. This is an attempt to clarify (@nquesada)

Reply:
Also @thisac!